syntax = "proto2";

package tec.proto;

message MovementCommand {
	optional bool forward = 1;
	optional bool backward = 2;
	optional bool leftStrafe = 3;
	optional bool rightStrafe = 4;
}

message OrientationCommand {
	required float x = 1;
	required float y = 2;
	required float z = 3;
	required float w = 4;
}

message ClientCommands {
	required uint64 id = 1;
	required uint64 commandID = 2;
	optional uint64 lastStateID = 3;
	optional MovementCommand movement = 4;
	optional OrientationCommand orientation = 5;
	repeated string commandList = 6;
}

message ChatCommand {
	required string command = 1;
	repeated string arguments = 2;
}

message UserLogin {
	required string username = 1;
	required string password = 2;
}
